mmedit/datasets/pipelines/crop.py

```diff
@@ -39,6 +45,21 @@ def _crop(self, data):
         for item in data_list:
             data_h, data_w = item.shape[:2]
             crop_h, crop_w = self.crop_size
+
+            if crop_h > data_h:
+                crop_y_offset = abs(data_h - crop_h) // 2
+                pad_width = ((2 * crop_y_offset, 2 * crop_y_offset), (0, 0),
+                             (0, 0))
+                item = np.pad(
+                    item, pad_width, mode='constant', constant_values=0)
+            elif crop_w > data_w:
+                crop_x_offset = abs(data_w - crop_w) // 2
+                pad_width = ((0, 0), (2 * crop_x_offset, 2 * crop_x_offset),
+                             (0, 0))
+                item = np.pad(
+                    item, pad_width, mode='constant', constant_values=0)
+
+            data_h, data_w = item.shape[:2]
```

Comment:
@ckkelvinchan just realised one thing: the original Crop class simply keeps the original shape of the image if crop_size is bigger than data_size, whereas in the modified Crop class the image is zero-padded. Whilst this works fine for the AOt-GAN implementation, I'm not sure if it might raise an error in other models in mmediting. Plus the unit test test_crop fails for random crop for size larger than the original shape. Wonder if it would be better to write a new class CenterCrop to take care of the zero-padding instead? Thanks

Reply:
![Screenshot from 2021-12-23 11-33-13](https://user-images.githubusercontent.com/33197366/147184134-de028bac-97bc-46a8-a96f-ce211459098a.png)
I have attached a screenshot for the test_crop unittest for your reference

Reply:
In the original Crop class, the assertion is that if crop_size is bigger than image size then the resulting image is resized such that it has the same shape as the original image.

Reply:
Or a better idea: we can add a parameter pad_zeros (bool) to Crop to give users the option to pad zeros depending on the model?

Reply:
> Or a better idea: we can add a parameter pad_zeros (bool) to Crop to give users the option to pad zeros depending on the model?

Yes, I think that's a good way to go. We can use a variable named `is_pad_zeros` to control it. But please assign the default value to `False` to avoid BC-breaking.